packages/gatsby-link/src/index.js

```diff
@@ -77,6 +78,13 @@ class GatsbyLink extends React.Component {
     }
   }
 
+  componentWillUnmount() {
+    const { instance, el } = this.io
```

Comment:
Let us make sure this.io is set.

```js
if (!this.io) { return }
```

Reply:
Ohh yes, forgot about that. Fixed in 5b827ab!